src/components/TourGuideProvider.tsx

```diff
@@ -133,15 +140,36 @@ export const TourGuideProvider = ({
     })
   }
 
-  const setCurrentStep = (key: string, step?: IStep) =>
-    new Promise<void>((resolve) => {
-      updateCurrentStep((currentStep) => {
-        const newStep = { ...currentStep }
-        newStep[key] = step
-        eventEmitter[key]?.emit('stepChange', step)
-        return newStep
-      })
-      resolve()
+  const setCurrentStep = async (key: string, step?: IStep) =>
+    new Promise<void>(async (resolve) => {
+      if (scrollRef && step) {
+        await step.wrapper.measureLayout(
```

Comment:
It's probably due to the age of the PR and the differences in react-native since then, but this needs to be `step.target`, or this will throw with the following error if someone wraps a composite component with a `TourGuideZone`: `ERROR  Warning: ref.measureLayout must be called with a ref to a native component.`